### DISCUSSION TITLE
Using Spectral API, how to use URL instead of OpenApi document and how to define the output format?

### DISCUSSION THREAD

**Discussion by P0lip** [stoplightio/spectral#1475] [Q&A]
> Originally posted in #1469

I want to use [Spectral Javascript API](https://meta.stoplight.io/docs/spectral/docs/guides/3-javascript.md), and I want to ask on how:
- to use `URL` instead of `OpenApiDocument`, for example to do this `spectral.run(myURL).then(console.log);`
- to define the **output format** like we can do with **Spectral CLI** ( `--format html` ).

Thank you!

**Comment by P0lip** [comment]
You can try using the following piece of code.

```ts
import { Document, isOpenApiv2, isOpenApiv3, Parsers, Spectral } from '@stoplight/spectral';
import { html } from '@stoplight/spectral/dist/cli/formatters';
import { DiagnosticSeverity } from '@stoplight/types';

fetch(myURL)
  .then(res => {
    if (!res.ok) {
      throw new Error('some error occurred'); // can be replaced with any other error handling you need
    }

    return res;
  })
  .then(res => res.text())
  .then(text => {
    const doc = new Document(text, Parsers.Json, myURL); // or Parsers.Yaml if you deal with Yaml
    const spectral = new Spectral();
    spectral.registerFormat('oas2', isOpenApiv2);
    spectral.registerFormat('oas3', isOpenApiv3);
    return spectral.loadRuleset('spectral:oas').then(() => spectral.run(doc));
  })
  .then(results => {
    return html(results, {
      failSeverity: DiagnosticSeverity.Warning, // or you can use some other setting, depending on the severity you like
    });
  });
```

**Reply by ghost** [reply]
Thanks a lot @P0lip , this is what I was looking for ☺️

Just later on if I want to return just **error messages** with the default format (instead of HTM format) with `DiagnosticSeverity.Error`, how can I replace this return code ?

``` javascript
.then(results => {
    return html(results, {
      failSeverity: DiagnosticSeverity.Warning, // or you can use some other setting, depending on the severity you like
    });
  });
```

**Reply by P0lip** [reply]
Which format do you mean by saying default format? Stylish?

If you have stylish in mind, you can do the following
```ts
import { stylish } from '@stoplight/spectral/dist/cli/formatters';

// rest of code
  .then(results => {
    return stylish(results, {
      failSeverity: DiagnosticSeverity.Warning, // or you can use some other setting, depending on the severity you like
    });
  });
```

**Reply by ghost** [reply]
thanks for coming back to me! 

I already tried that, what I want to say by default format is the format I got when using the javascript API, it's just an array of Json objects(it's different to the default format from the CLI which is `stylish`).

I want to use `DiagnosticSeverity` with this `javascript` code for example:

```JavaScript
spectral.loadRuleset("spectral:oas")
                .then(() => spectral.run(doc))
                .then(results => {
                    callback(null, results); //I want to use DiagnosticSeverity here
                });
```

**Reply by P0lip** [reply]
I'm not sure I understand the question.
`DiagnosticSeverity` is the level of severity of a given linting entry result.
Do you want to pass a string (formatted output) to callback instead of an array of results?
